transport/transport.go

```diff
@@ -330,12 +330,20 @@ func NewServerTransport(protocol string, conn net.Conn, maxStreams uint32, authI
 	return newHTTP2Server(conn, maxStreams, authInfo)
 }
 
+// DefaultDialerFn is the default dialer.
+func DefaultDialerFn(addr string, timeout time.Duration, cancel <-chan struct{}) (net.Conn, error) {
```

Comment:
1. make these function unexported;
2. yes, write the function you want to use in the tests. end2end tests should behave like how a normal user uses grpc (notice this user does not know transport package).